### DISCUSSION TITLE
Create New Service with Custom UI

### DISCUSSION THREAD

**Discussion by benlevinNEU** [RavenSystem/esp-homekit-devices#1883] [Q&A] [answered]
Hi, I'm trying to build a custom HomeKit compatible service using the Arduino ESP8266. I've been successful in making a few very simple devices but now I'm trying to build a controller for a light strip that has a number of custom characteristics. I want:

* Name
* On\off state (bool)
* Play\pause state (bool)
* Brightness (int)
* Speed (int)
* Mode (char)

I'm not entirely sure which characteristics I can repurpose for the play\pause, speed and mode or if I can create new ones that have their own UI elements.

How can I create a custom service that supports all of these characteristics?

Is this even possible?

I'm trying to build a UI that looks kind of like this except with sliders for brightness and speed. I built this UI for my existing implementation of the project which runs a webserver on the esp and you can query it from your phone.

![image](https://user-images.githubusercontent.com/73363889/190434395-0bf058a5-c7e1-4f77-adf7-e81fcafd6878.png)

**Comment by RavenSystem** [comment, marked as answer]
This site is for HAA firmware questions.
